src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -834,3 +835,10 @@ def type_string_to_type_object(type_str: str) -> DataType:
 ColumnOrSqlExpr = Union["snowflake.snowpark.column.Column", str]
 LiteralType = Union[VALID_PYTHON_TYPES_FOR_LITERAL_VALUE]
 ColumnOrLiteral = Union["snowflake.snowpark.column.Column", LiteralType]
+
+class CopyOptions(TypedDict, total=False):
+    overwrite: bool
+    single: bool
+    max_file_size: float
+    include_query_id: bool
+    detailed_output: bool
```

Comment:
This allows the IDE to tell you when it receives unexpected keyword arguments.

![image](https://github.com/snowflakedb/snowpark-python/assets/71691411/7e76129d-0e82-4376-bcfd-d7f600d98161)